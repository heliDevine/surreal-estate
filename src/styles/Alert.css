.Alert {
    border-color: red;
    border-color: green;
}
/* .Alert {
    padding: 5px;
    margin-bottom: 5px;
  }
  
  .Alert:empty {
    display: none;
  }
  
  .alert-success {
    border: 1px solid lightgreen;
    background-color: lightgreen;
    color: green;
  }
  
  .alert-error {
    border: 1px solid lightpink;
    background-color: lightpink;
    color: red;
  } */
