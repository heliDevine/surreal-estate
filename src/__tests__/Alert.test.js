import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../components/Alert';

// it('displays an error message', () => {
//     const component = render(<Alert message="Error!" />);
//     const alertMessageNode = component.getByText('Error!');
//     expect(alertMessageNode.textContent).toBe('Error!');
// });
it('displays an error message', () => {
    const { getByText } = render(<Alert message="Error!" />);

    expect(getByText(/Error/).textContent).toBe('Error!');
});
it('displays a success message', () => {
    const { getByText } = render(<Alert message="Success!!!!" success />);
    expect(getByText(/Success/).textContent).toBe('Success!!!!');
});
it('does not render an error or a success message if message props is empty', () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
});

it('renderds error message when there is an error', () => {
    const { asFragment } = render(<Alert message="error" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
});
it('renders success message when property has been created', () => {
    const { asFragment } = render(<Alert message="Property Added" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
});
