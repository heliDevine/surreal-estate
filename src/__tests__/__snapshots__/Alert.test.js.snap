// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `<DocumentFragment />`;

exports[`renderds error message when there is an error 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    error
  </div>
</DocumentFragment>
`;

exports[`renders success message when property has been created 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Property Added
  </div>
</DocumentFragment>
`;
